lib/ueberauth/strategy/slack.ex

```diff
@@ -133,23 +134,32 @@ defmodule Ueberauth.Strategy.Slack do
     user = conn.private[:slack_user]
     auth = conn.private[:slack_auth]
     identity = conn.private[:slack_identity]
-    image_urls = (get_in(user, ["profile"]) || %{})
+    profile = (get_in(user, ["profile"]) || get_in(identity, ["user"]) || %{})
+    image_urls = profile
     |> Map.keys
     |> Enum.filter(&(&1 =~ ~r/^image_/))
-    |> Enum.map(&({&1, user["profile"][&1]}))
+    |> Enum.map(&({&1, profile[&1]}))
+    |> Enum.into(%{})
+
+    team_profile = (get_in(identity, ["team"]) || %{})
+    team_image_urls = team_profile
```

Comment:
This pipeline does a lot that and goes through the map a number of times, what do you think about refactoring this to use `Enum.reduce/3`?

```elixir
identity
|> Map.get("team", %{})
|> Enum.reduce(%{}, fn {key, value}, acc -> 
  if String.match?(key, ~r/^image_/) do
    Map.put(acc, "team_#{key}", value)
  else
    acc
  end
end)
```

Reply:
Maybe something like this is a bit more simple
```
identity
|> Map.get("team", %{}) 
|> Enum.filter(fn {key, value} -> (key =~ ~r/^image_/) end)
|> Enum.into(%{}, fn {key, value} -> {"team_#{key}", value}) end)
```